src/main/java/Duke.java

```diff
@@ -1,10 +1,181 @@
+
+                if (curTask.equals("mark") || curTask.equals("unmark")) {
+                    int indexOfSpace = echo.indexOf(" ");
+                    int indexOfTask = Integer.parseInt(echo.substring(indexOfSpace + 1));
+                    if ((curTask.equals("mark") && (tasks.get(indexOfTask - 1).isDone)) ||
+                        (curTask.equals("unmark") && (!tasks.get(indexOfTask - 1).isDone))) {
+                        throw new ChangeStatusException("This task is already marked/unmarked");
+                    }
+                    toggleStatus(tasks, indexOfTask);
+                    writeToFile(tasks);
+                    continue;
+                } else if (curTask.equals("deadline")) {
+                    addDeadline(echo, tasks, index);
+                } else if (curTask.equals("event")) {
+                    addEvent(echo, tasks, index);
+                } else if (curTask.equals("todo")) {
+                    addTodo(echo, tasks, index);
+                } else if (curTask.equals("delete")) {
+                    deleteTask(tasks, index);
+                    index = index - 2;
+                }
+                index++;
+                writeToFile(tasks);
+                System.out.print("Now you have ");
+                System.out.println(index + " tasks in the list.");
+                System.out.println("----------------------------------------------------------------");
+
+            } catch (StringIndexOutOfBoundsException e) {
+                System.out.println("☹ OOPS!!! I'm sorry, but I don't know what that means :-( ");
+                System.out.println("----------------------------------------------------------------");
+            } catch (NullPointerException e) {
+                System.out.println("☹ OOPS!!! I'm sorry, but seems like there is no such task :-( ");
+                System.out.println("----------------------------------------------------------------");
+            } catch (ChangeStatusException e) {
+                System.out.println("☹ OOPS!!! I'm sorry, but seems like this task is already marked/unmarked");
+                System.out.println("----------------------------------------------------------------");
+            } catch (DukeException e) {
+                System.out.print("☹ OOPS!!! The description of a ");
+                System.out.print(e);
+                System.out.println(" cannot be empty");
+                System.out.println("----------------------------------------------------------------");
+            }
+        }
+    }
+
+    private static void writeToFile(ArrayList<Task> tasks) throws IOException {
+        FileWriter fw = new FileWriter("duke.txt");
+        for(Task t: tasks){
+            fw.write(t.toString()+"\n");
+        }
+        fw.close();
+    }
+
+    private static void printFileContents(String filePath) throws IOException, FileNotFoundException {
+        File f = new File(filePath); // create a File for the given file path
```

Comment:
[Writing comment](https://se-education.org/guides/conventions/java/basic.html#comments) in a separate line with same level of indentation as code will ensure that the logical structure of code is not broken.